include/utils/FFTData.h

```diff
@@ -63,3 +77,15 @@ class FFTData
   /// FFT data buffer
   std::vector<T> _buffer;
 };
+
+template <typename T>
+template <typename T1>
+void
+FFTData<T>::applyLambdaReciprocalSpace(T1 lambda,
+                                       const std::vector<Real> & ktable0,
+                                       const std::vector<Real> & ktable1,
+                                       const std::vector<Real> & ktable2)
+{
+  for (size_t index = 0; index < ktable0.size() * ktable1.size() * ktable2.size(); index++)
+    _buffer[index] = lambda(index);
+}
```

Comment:
my suggestion involved passing in the k-vector to the lambda. Then you can replace some more loops with lambdas. But we can do that later.

Reply:
passing in ktable seems unnecessary in your variant of the function if you can just evaluate `_buffer.size()`